rclcpp_lifecycle/src/lifecycle_node.cpp

```diff
@@ -152,8 +152,19 @@ LifecycleNode::LifecycleNode(
 
 LifecycleNode::~LifecycleNode()
 {
+  auto current_state = LifecycleNode::get_current_state().id();
+  if (current_state != lifecycle_msgs::msg::State::PRIMARY_STATE_FINALIZED) {
+    // This might be leaveing sensors and devices without shutting down unintentionally.
+    // It is user's responsibility to call shutdown to avoid leaving them unknow states.
+    RCLCPP_DEBUG(
+      rclcpp::get_logger("rclcpp_lifecycle"),
+      "LifecycleNode is not shut down: Node still in state(%u) in destructor",
+      current_state);
+  }
+
```

Comment:
i am not 100% confident to have this debug print here, would be useful to let the user know during debug mode. i would like to have feedback here.

Reply:
I would suggest a warning, rather than a debug log.
This is something users should notified about and I don't expect it to be too spammy (as it just logs 1 line when you destroy nodes)

Reply:
that is true. but i was thinking that is kinda surprising when this warning will just pop up if the user application shuts down the `LifecycleNode` in the application... i am okay to switch it back into `warning`. i will wait for another comment here.

Reply:
ended up changing that into `WARNING`, that would be more useful for user application. see https://github.com/ros2/rclcpp/pull/2562/commits/3ad7181de12817cbef965ec22747ec525cefd757